scipy/stats/_stats_py.py

```diff
@@ -4544,11 +4571,19 @@ def confidence_interval(self, confidence_level=0.95):
         .. [1] "Pearson correlation coefficient", Wikipedia,
                https://en.wikipedia.org/wiki/Pearson_correlation_coefficient
         """
-        return _pearsonr_fisher_ci(self.statistic, self._n, confidence_level,
-                                   self._alternative)
-
+        if isinstance(method, BootstrapMethod):
+            ci = _pearsonr_bootstrap_ci(confidence_level, method,
+                                        self._x, self._y)
+        elif method is None:
+            ci = _pearsonr_fisher_ci(self.statistic, self._n, confidence_level,
+                                     self._alternative)
+        else:
+            message = ('`method` must be an instance of `BootstrapMethod` '
+                       'or None.')
+            raise ValueError(message)
+        return ci
 
-def pearsonr(x, y, *, alternative='two-sided'):
+def pearsonr(x, y, *, alternative='two-sided', method=None):
```

Comment:
I think `method` is ok in the context of `testXXX.confidence_interval`, but for the test itself it could be confusing. Here without reading the docstrings I could think this is to specify a "version" or "alternative" to the computation itself. Well it is somehow the case, but I think it has a different meaning for a user.

Instead I think `resampling_method` would be more appropriate. It would also match your base class which would make it clearer in the code and help maintenance.